commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java

```diff
@@ -0,0 +1,171 @@
+        }
+        return !defaultPrefix;
+    }
+
+    @Override
+    public boolean trigger(CodeGenContext context) {
+        String prefix = computeConnectClientPrefix(context.config());
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = context.inputDir().resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            String content = null;
+            try {
+                content = Files.readString(customClient);
+            } catch (IOException e) {
+                LOGGER.debug(
+                        "Cannot read content of custom connect-client.ts ({}). File will not be overwritten.",
+                        customClient);
+            }
+            if (content != null) {
+                content = content.replaceFirst(
+                        "(const client = new ConnectClient\\(\\{prefix:\\s*')[^']+('}\\);)", "$1" + prefix + "$2");
+                try {
+                    Files.writeString(customClient, content);
+                    LOGGER.debug(
+                            "Prefix in custom connect-client.ts ({}) replaced with new value {}", customClient, prefix);
+                    return true;
+                } catch (IOException e) {
+                    LOGGER.debug("Cannot write content of custom connect-client.ts ({}).", customClient);
+                }
+            }
+        } else if (!defaultPrefix) {
+            return writeConnectClient(prefix, customClient);
+        }
+        return false;
+    }
+
+    private static String computeConnectClientPrefix(Config config) {
+        String prefix = config.getValue("vaadin.endpoint.prefix", String.class);
+        if (prefix.startsWith("/")) {
+            prefix = prefix.substring(1);
+        }
+        if (prefix.endsWith("/")) {
+            prefix = prefix.substring(0, prefix.length() - 1);
+        }
+        return prefix;
+    }
+
+    static boolean writeConnectClient(String prefix, Path customClient) {
+        InputStream template = Endpoint.class.getResourceAsStream("/connect-client.default.template.ts");
+        if (template != null) {
+            try (InputStream is = template;
+                    Scanner scanner = new Scanner(is).useDelimiter("\\A")) {
+                if (scanner.hasNext()) {
+                    String out = scanner.next().replace("{{PREFIX}}", prefix);
+
+                    Files.writeString(customClient, out, StandardCharsets.UTF_8);
+                    return true;
+                } else {
+                    LOGGER.debug("Template file connect-client.default.template.ts is empty.");
+                }
+            } catch (IOException ex) {
+                LOGGER.debug("Cannot read template file connect-client.default.template.ts.", ex);
+            }
+        } else {
+            LOGGER.debug("Cannot find template file connect-client.default.template.ts.");
+        }
+        return false;
+    }
```

Comment:
I would suggest guard clauses here for better readability.

Reply:
Can you please elaborate? Just to be sure I got your point

Reply:
done